### ISSUE TITLE
Error message on transferring shard

### ISSUE THREAD

**Issue by generall** [qdrant/qdrant#2804]
## Current Behavior

When I try to `move_shard` from one node to another, I can see a error message:

```
2023-10-12T13:22:53.806459Z ERROR qdrant::tonic::logging: gRPC /qdrant.CollectionsInternal/Initiate unexpectedly failed with Internal error "Service internal error: Cannot un-proxify local shard 0 on peer 996237883537134 because it is not active" 0.237796    
```

or

```
2023-10-12T13:23:21.308463Z ERROR collection::shards::transfer::shard_transfer: Failed to transfer shard 0 -> 5160243291937329: Service internal error: Internal error: status: Internal, message: "Service internal error: Cannot un-proxify local shard 0 on peer 5160243291937329 because it is not active", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Thu, 12 Oct 2023 13:23:20 GMT", "content-length": "0"} }    
```

Even though, the transfer itself is eventually completed successfully, the error might indicate that something is not working exactly as expected.

## Steps to Reproduce

1. Create 2 nodes in a cluster
2. Create small collection with 2 shards and 1 replica
3. Trigger shard transfer from one node to another 
4. Check logs of the reciever

## Expected Behavior

No error messages

## Possible Solution

It is strange, that receiver node tries to un-proxify something. Proxy is usually created on the transmitter side.
So maybe it worth to start investigation with this

**Comment by timvisee** [comment]
My findings thus far:

The current implementation of transferring shards is very time sensitive. Imagine having two nodes, A and B, where a shard is transferred from A -> B.

This is what should happen:
1. A applies the shard transfer to consensus.
2. B accepts the shard transfer from consensus and prepares locally.
3. A starts sending the shard to B.
4. B starts receiving the shard from A.
5. A and B finish the transfer and agree on this through consensus.

This is what currently happens:

1. A applies the shard transfer to consensus.
2. **A starts sending the shard to B.**
3. **B starts receiving the shard from A.**
4. **B accepts the shard transfer from consensus and prepares locally.**
5. A and B finish the transfer and agree on this through consensus.

In other words, node A starts the transfer too quickly, before B agreed with this operation. Sending the shard requires preparation on node B, but that node did not have time to prepare yet.

Adding a delay of 1 second [here](https://github.com/qdrant/qdrant/blob/fa3aebe6608bc62eb2c25589713679d8ce9b3afa/lib/collection/src/shards/transfer/shard_transfer.rs#L464), just before node A starts transferring to node B, prevents the issue from happening.

**Comment by ffuugoo** [comment]
Seems like we broke `initiate_shard_transfer` when we were fixing another bug regarding shard transfers in #2373. 😂

**Comment by timvisee** [comment]
Fixed by #2813
